contrib_bots/lib/giveaway.py

```diff
@@ -0,0 +1,211 @@
+# See readme.md for instructions on running this code.
+from __future__ import print_function
+import time
+import threading
+import random
+from past.utils import old_div
+
+class GiveawayHandler(object):
+    '''
+    This plugin allows for "giveaways" to be run within a
+    zulip stream. The bot will look for any message starting
+    with "@giveaway" followed by a command (either init, enter,
+    exit or cancel) followed by a time in seconds for the giveaway
+    to run for. Full syntax is shown below:
+    @giveaway command giveaway_name giveaway_time
+
+    Possible commands:
+    init - initialize a giveaway (only one can be running at a time)
+    enter - enter a giveaway (a giveaway must be running to use this command)
+    exit - leave a giveaway (must be partaking in an active giveaway for this
+    command to work
+    cancel - cancel a giveaway (can only be run by the person who first
+    initialised the giveaway
+
+    This plugin was created by Daniel O'Brien
+    as part of Google Code-In 2016-2017
+    '''
+    def __init__(self):
+        self.giveawayArray = []
+        self.giveawayOn = False
+        self.giveawayName = ""
+        self.giveawayTime = 0
+        self.giveawayCreator = ""
+        self.giveawayStartTime = 0
+        self.giveawayGroup = ""
+        self.giveawaySubject = ""
+        self.giveawayClient = ""
+        self.giveawayTimeMins = 0
+        self.giveawayStage = 0
+        self.one_minute_warning = False
+        self.giveawayWinner = ""
+        self.winnerContactDetails = []
+        self.winnerIndex = 0
+
+    def time_update(self, new_content):
+            if self.giveawayOn == True:
```

Comment:
`==` is not the good comparator.
Do: 
`if self.giveawayOn` or if it makes sense in the case `if self.giveawayOn is True`